catalogue/webapp/components/ImageEndpointSearchResults/ImageEndpointSearchResults.tsx

```diff
@@ -71,6 +80,55 @@ const ImageEndpointSearchResults: FunctionComponent<Props> = ({
   const { isFullSupportBrowser } = useContext(AppContext);
   const [expandedImage, setExpandedImage] = useState<Image | undefined>();
   const [isActive, setIsActive] = useState(false);
+  const router = useRouter();
+  const toggles = useToggles();
+
+  const imageMap: Record<string, Image> = images.reduce(
+    (a, image) => ({ ...a, [image.id]: image }),
+    {}
+  );
+
+  const setImageIdInURL = (id: string) => {
+    window.location.hash = id;
+  };
+
+  useEffect(() => {
+    const onHashChanged = async () => {
+      const hash = window.location.hash.slice(1);
+      if (!hash) {
+        setIsActive(false);
+        setExpandedImage(undefined);
+      }
+      // see if the new hash is already in the imageMap
+      if (imageMap[hash]) {
+        // if it is, update the expanded image
+        setExpandedImage(imageMap[hash]);
+        setIsActive(true);
+      } else {
+        // if it's not, fetch the image and then update
+        const { image } = await getImage({ id: hash, toggles });
+        if (image.type === 'Image') {
+          imageMap[image.id] = image;
+          setExpandedImage(image);
+        }
+      }
+    };
+    onHashChanged();
+    window.addEventListener('hashchange', onHashChanged);
+    return () => {
+      window.removeEventListener('hashchange', onHashChanged);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (isActive) {
+      setImageIdInURL(expandedImage?.id || '');
+    } else {
+      router.push(router.asPath, undefined, {
+        shallow: true,
```

Comment:
Is this the router pushing its own path? What does that achieve? A comment explaining why we're doing that, and why we're using `shallow: true`, would be good here

Reply:
yes, this clears the fragment entirely, and not just leaving the # with no comment